app/V1Module/presenters/ShadowAssignmentsPresenter.php

```diff
@@ -0,0 +1,377 @@
+
+      // create all new localized texts
+      $externalAssignmentLink = trim(Arrays::get($localization, "link", ""));
+      if ($externalAssignmentLink !== "" && !Validators::isUrl($externalAssignmentLink)) {
+        throw new InvalidArgumentException("External assignment link is not a valid URL");
+      }
+
+      $localized = new LocalizedShadowAssignment(
+        $lang, $localization["name"], $localization["text"],
+        $externalAssignmentLink ?: null
+      );
+
+      $localizedTexts[$lang] = $localized;
+    }
+
+    // make changes to database
+    Localizations::updateCollection($assignment->getLocalizedTexts(), $localizedTexts);
+
+    foreach ($assignment->getLocalizedTexts() as $localizedText) {
+      $this->shadowAssignments->persist($localizedText, false);
+    }
+
+    $this->shadowAssignments->flush();
+    $this->sendSuccessResponse($this->shadowAssignmentViewFactory->getAssignment($assignment));
+  }
+
+  /**
+   * Create new shadow assignment in given group.
+   * @POST
+   * @Param(type="post", name="groupId", description="Identifier of the group")
+   * @throws ForbiddenRequestException
+   * @throws BadRequestException
+   * @throws NotFoundException
+   */
+  public function actionCreate() {
+    $req = $this->getRequest();
+    $group = $this->groups->findOrThrow($req->getPost("groupId"));
+
+    if (!$this->groupAcl->canCreateShadowAssignment($group)) {
+      throw new ForbiddenRequestException("You are not allowed to create assignment in given group.");
+    }
+
+    if ($group->isOrganizational()) {
+      throw new BadRequestException("You cannot create assignment in organizational groups");
+    }
+
+    $assignment = ShadowAssignment::assignToGroup($group);
+    $this->shadowAssignments->persist($assignment);
+    $this->sendSuccessResponse($this->shadowAssignmentViewFactory->getAssignment($assignment));
+  }
+
+  public function checkRemove(string $id) {
+    $assignment = $this->shadowAssignments->findOrThrow($id);
+
+    if (!$this->shadowAssignmentAcl->canRemove($assignment)) {
+      throw new ForbiddenRequestException("You cannot remove this shadow assignment.");
+    }
+  }
+
+  /**
+   * Delete shadow assignment
+   * @DELETE
+   * @param string $id Identifier of the assignment to be removed
+   * @throws NotFoundException
+   */
+  public function actionRemove(string $id) {
+    $assignment = $this->shadowAssignments->findOrThrow($id);
+    $this->shadowAssignments->remove($assignment);
+    $this->sendSuccessResponse("OK");
+  }
+
+  public function checkEvaluations(string $id) {
+    $assignment = $this->shadowAssignments->findOrThrow($id);
+    if (!$this->shadowAssignmentAcl->canViewEvaluations($assignment)) {
+      throw new ForbiddenRequestException();
+    }
+  }
+
+  /**
+   * Get a list of evaluations of all users for the assignment
+   * @GET
+   * @param string $id Identifier of the assignment
+   * @throws NotFoundException
+   */
+  public function actionEvaluations(string $id) {
```

Comment:
What on earth does evaluation mean in this context? Have you considered naming it something like PointAward?